barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -3369,28 +3407,30 @@ std::vector<Row> AvmTraceBuilder::finalize(uint32_t min_trace_size, bool range_c
-            } else if (src.op_emit_nullifier) {
-                next.avm_kernel_emit_nullifier_write_offset = curr.avm_kernel_emit_nullifier_write_offset + 1;
-            } else if (src.op_nullifier_exists) {
-                next.avm_kernel_nullifier_exists_write_offset = curr.avm_kernel_nullifier_exists_write_offset + 1;
-            } else if (src.op_l1_to_l2_msg_exists) {
-                next.avm_kernel_l1_to_l2_msg_exists_write_offset = curr.avm_kernel_l1_to_l2_msg_exists_write_offset + 1;
-            } else if (src.op_emit_l2_to_l1_msg) {
-                next.avm_kernel_emit_l2_to_l1_msg_write_offset = curr.avm_kernel_emit_l2_to_l1_msg_write_offset + 1;
-            } else if (src.op_emit_unencrypted_log) {
-                next.avm_kernel_emit_unencrypted_log_write_offset =
-                    curr.avm_kernel_emit_unencrypted_log_write_offset + 1;
-            } else if (src.op_sload) {
-                next.avm_kernel_sload_write_offset = curr.avm_kernel_sload_write_offset + 1;
-            } else if (src.op_sstore) {
-                next.avm_kernel_sstore_write_offset = curr.avm_kernel_sstore_write_offset + 1;
+            if (src.op_emit_unencrypted_log) {
+                info("side effect counter when emiting log: ", curr.avm_kernel_side_effect_counter);
             }
 
+            // Increment the write offset counter for the following row
+            next.avm_kernel_note_hash_exist_write_offset =
+                curr.avm_kernel_note_hash_exist_write_offset + static_cast<const int>(src.op_note_hash_exists);
+            next.avm_kernel_emit_note_hash_write_offset =
+                curr.avm_kernel_emit_note_hash_write_offset + static_cast<const int>(src.op_emit_note_hash);
+            next.avm_kernel_emit_nullifier_write_offset =
+                curr.avm_kernel_emit_nullifier_write_offset + static_cast<const int>(src.op_emit_nullifier);
+            next.avm_kernel_nullifier_exists_write_offset =
+                curr.avm_kernel_nullifier_exists_write_offset + static_cast<const int>(src.op_nullifier_exists);
+            next.avm_kernel_l1_to_l2_msg_exists_write_offset =
+                curr.avm_kernel_l1_to_l2_msg_exists_write_offset + static_cast<const int>(src.op_l1_to_l2_msg_exists);
+            next.avm_kernel_emit_l2_to_l1_msg_write_offset =
+                curr.avm_kernel_emit_l2_to_l1_msg_write_offset + static_cast<const int>(src.op_emit_l2_to_l1_msg);
+            next.avm_kernel_emit_unencrypted_log_write_offset =
+                curr.avm_kernel_emit_unencrypted_log_write_offset + static_cast<const int>(src.op_emit_unencrypted_log);
+            next.avm_kernel_sload_write_offset =
+                curr.avm_kernel_sload_write_offset + static_cast<const int>(src.op_sload);
+            next.avm_kernel_sstore_write_offset =
+                curr.avm_kernel_sstore_write_offset + static_cast<const int>(src.op_sstore);
+
```

Comment:
We increment the write offset counter so that the next kernel lookup in the main trace is constrained to the proper entry in the kernel trace?

Reply:
yep